src/main/java/org/folio/rest/impl/RelatedInstanceAPI.java

```diff
@@ -0,0 +1,100 @@
+package org.folio.rest.impl;
+
+import java.util.Map;
+
+import javax.validation.constraints.Pattern;
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.jaxrs.model.RelatedInstance;
+import org.folio.rest.jaxrs.model.RelatedInstances;
+import org.folio.rest.persist.PgUtil;
+import org.folio.services.relatedInstance.RelatedInstanceService;
+import static org.folio.rest.support.EndpointFailureHandler.handleFailure;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+/**
+ * Implements the related instance persistency using postgres jsonb.
+ */
+public class RelatedInstanceAPI implements org.folio.rest.jaxrs.resource.InstanceStorageRelatedInstances {
+  public static final String RELATED_INSTANCE_TABLE = "related_instance";
+
+  @Override
+  public void getInstanceStorageRelatedInstances(String query, int offset, int limit, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.get(RELATED_INSTANCE_TABLE, RelatedInstance.class, RelatedInstances.class, query, offset, limit,
+      okapiHeaders, vertxContext, GetInstanceStorageRelatedInstancesResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void postInstanceStorageRelatedInstances(String lang, RelatedInstance entity, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    if (checkInstanceRelatedToSelf(entity)) {
+      asyncResultHandler.handle(cannotBeRelatedToSelf());
+      return;
+    }
+
+    new RelatedInstanceService(vertxContext, okapiHeaders)
+      .createRelatedInstance(entity)
+      .onSuccess(response -> asyncResultHandler.handle(Future.succeededFuture(response)))
+      .onFailure(handleFailure(asyncResultHandler));
+  }
+
+  @Override
+  public void getInstanceStorageRelatedInstancesByRelatedInstanceId(String instanceId, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.getById(RELATED_INSTANCE_TABLE, RelatedInstance.class, instanceId,
+        okapiHeaders, vertxContext, GetInstanceStorageRelatedInstancesByRelatedInstanceIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void deleteInstanceStorageRelatedInstancesByRelatedInstanceId(String instanceId, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.deleteById(RELATED_INSTANCE_TABLE, instanceId, okapiHeaders, vertxContext,
+      DeleteInstanceStorageRelatedInstancesByRelatedInstanceIdResponse.class, asyncResultHandler);
+  }
```

Comment:
Does this delete all related instance records related to a specific instance?

How would a client only delete a single relationship between two instances?

Reply:
> I think the use of "instance" is throwing you here. The instanceId being used here is the UUID assigned to the relationship itself, not the instance records involved. 

That is likely. I find that confusing and a little misleading. What controls that naming?

bq. In other words, the ID of an instance of a relationship.

Hmm, I find that terminology confusing.

bq. I worried this might be confusing, but the other crud operations and endpoints all use "instanceID" in this context to refer to the UUID of a specific data entry, so there's a consistency issue as well.

Which other CRUD endpoints are you referring to?

Reply:
Hrm.  Well, on checking again, it seems that naming of the UUID in update/delete functions isn't as uniform as I thought at first.  I'll change the name of the element to "id"

Reply:
Thank you

Reply:
clarification:  are you asking me to change the name of the instanceId parameter, or the name of the function (deleteInstanceStorageRelatedInstancesByRelatedInstanceId)?  The latter I can change easily, the former is based on the name of the element in the RAML.  the "byinstanceId" part is auto-generated by the code that stubs out the CRUD methods.